backend/src/data/repositories/geolocation-repository.ts

```diff
@@ -0,0 +1,23 @@
+import { GeolocationModel } from '../models';
+import { IGeolocation } from '~/common/interfaces';
+
+class GeolocationRepository {
+  public createGeolocation(geolocation: IGeolocation): Promise<IGeolocation> {
+    return GeolocationModel.create(geolocation);
+  }
+
+  public getByUserId(userId: string): Promise<IGeolocation | null> {
+    return GeolocationModel.findOne({ where: { userId } });
+  }
+
+  public async updateById(id: string, data: IGeolocation): Promise<IGeolocation> {
+    const result = await GeolocationModel.update(data, {
+      where: { id },
+      returning: true,
+    });
+
+    return result[1][0];
+  }
```

Comment:
```suggestion
  public async updateById(id: string, data: IGeolocation): Promise<IGeolocation> {
    const [, updatedArticle] = await GeolocationModel.update(data, {
      where: { id },
      returning: true,
    });

    return updatedArticle;
  }
```
Should be something like this -